src/fileimpl.cpp

```diff
@@ -111,36 +111,45 @@ sectionSubReader(const FileReader& zimReader, const std::string& sectionName,
 
     quickCheckForCorruptFile();
 
-    offset_t titleOffset(header.getTitleIdxPos());
-    zsize_t  titleSize(4*header.getArticleCount());
-
-    for (auto url:{"listing/titleOrdered/v1", "listing/titleOrdered/v0"}) {
-      auto result = direntLookup().find('W', url);
-      if (result.first) {
-        auto dirent = mp_urlDirentAccessor->getDirent(result.second);
-        auto cluster = getCluster(dirent->getClusterNumber());
-        if (cluster->isCompressed()) {
-          // This is a ZimFileFormatError.
-          // Let's be tolerent and skip the entry
-          continue;
-        }
-        titleOffset = getClusterOffset(dirent->getClusterNumber()) + offset_t(1) + cluster->getBlobOffset(dirent->getBlobNumber());
-        titleSize = cluster->getBlobSize(dirent->getBlobNumber());
-        break;
-      }
+    mp_titleDirentAccessor = getTitleAccessor("listing/titleOrdered/v1");
+
+    if (!mp_titleDirentAccessor) {
+      offset_t titleOffset(header.getTitleIdxPos());
+      zsize_t  titleSize(4*header.getArticleCount());
+      mp_titleDirentAccessor = getTitleAccessor(titleOffset, titleSize, "Title index table");
     }
 
-    auto titleIndexReader = sectionSubReader(*zimReader,
-                                             "Title index table",
-                                             titleOffset,
-                                             titleSize);
+    readMimeTypes();
+  }
 
-    mp_titleDirentAccessor.reset(
-        new IndirectDirentAccessor(mp_urlDirentAccessor, std::move(titleIndexReader), title_index_t(titleSize.v/4)));
+  std::unique_ptr<IndirectDirentAccessor> FileImpl::getTitleAccessor(const std::string& path)
+  {
+    auto result = direntLookup().find('W', path);
+    if (!result.first) {
+      return nullptr;
+    }
 
-    readMimeTypes();
+    auto dirent = mp_urlDirentAccessor->getDirent(result.second);
+    auto cluster = getCluster(dirent->getClusterNumber());
+    if (cluster->isCompressed()) {
+      // This is a ZimFileFormatError.
+      // Let's be tolerent and skip the entry
+      return nullptr;
+    }
+    auto titleOffset = getClusterOffset(dirent->getClusterNumber()) + offset_t(1) + cluster->getBlobOffset(dirent->getBlobNumber());
```

Comment:
It would be nice of `zim::Cluster::getBlobOffset()` was in charge of the 1 (the cluster type byte) required in this computation of the full offset relative to the beginning of the ZIM archive. That would make it more consistent with `zim::Cluster::getOffset()`. Only a single pre-existing usage of `zim::Cluster::getBlobOffset()` in `FileImpl::getBlobOffset()` needs to be updated, in addition to this new code.